packages/core/src/store/pin-store-factory.ts

```diff
@@ -2,38 +2,45 @@ import { IpfsApi, PinningBackendStatic } from "@ceramicnetwork/common";
 import { LevelStateStore } from "./level-state-store";
 import { PinningAggregation } from "@ceramicnetwork/pinning-aggregation";
 import { PinStore } from "./pin-store";
+import { StateStore } from "./state-store";
 import CID from 'cids'
 import path from "path";
 import os from "os";
-import { promises as fs } from 'fs'
 import { IpfsPinning } from '@ceramicnetwork/pinning-ipfs-backend'
 
-const DEFAULT_PINSET_DIRECTORY = path.join(os.homedir(), ".ceramic", "pinset")
+const DEFAULT_STATE_STORE_DIRECTORY = path.join(os.homedir(), ".ceramic", "statestore")
 const IPFS_GET_TIMEOUT = 60000 // 1 minute
 
 export type Props = {
     networkName?: string;
-    pinsetDirectory?: string;
+    stateStoreDirectory?: string;
     pinningEndpoints?: string[];
     pinningBackends?: PinningBackendStatic[];
 }
 
 export class PinStoreFactory {
-    readonly stateStorePath: string
+    readonly localStateStoreDirectory: string
     readonly pinningEndpoints: string[]
-    readonly pinningBackends: PinningBackendStatic[];
+    readonly pinningBackends: PinningBackendStatic[]
+    readonly networkName: string
+    private _stateStore: StateStore
 
     constructor(readonly ipfs: IpfsApi, props: Props) {
-        const directoryRoot = props.pinsetDirectory || DEFAULT_PINSET_DIRECTORY
-        // Always store the pinning state in a network-specific directory
-        this.stateStorePath = path.join(directoryRoot, props.networkName)
+        this.networkName = props.networkName
+        this.localStateStoreDirectory = props.stateStoreDirectory || DEFAULT_STATE_STORE_DIRECTORY
         this.pinningEndpoints = props.pinningEndpoints && props.pinningEndpoints.length > 0 ? props.pinningEndpoints : ['ipfs+context']
         this.pinningBackends = props.pinningBackends && props.pinningBackends.length > 0 ? props.pinningBackends : [IpfsPinning]
     }
 
-    async open(): Promise<PinStore> {
-        await fs.mkdir(this.stateStorePath, { recursive: true }) // create dir if it doesn't exist
-        const stateStore = new LevelStateStore(this.stateStorePath)
+    public setStateStore(stateStore: StateStore): void {
+      this._stateStore = stateStore
+    }
+
+    public async open(): Promise<PinStore> {
```

Comment:
Seems like this method name should be `create`? Bc it creates and returns a new StateStore

Reply:
renamed to `createPinStore()`